testing/unittests/jlibtests.cpp

```diff
@@ -3731,6 +3732,58 @@ class PTreeSerializationDeserializationTest : public CppUnit::TestFixture
     {
         performRoundTripTest(__func__, createBinaryDataCompressionTestPTree(testXml));
     }
+
+    void testCreatePTreeWithNodeCreator()
+    {
+        // Test the createPTree(IBufferedSerialInputStream &src, const char *tag, IPTreeNodeCreator *nodeCreator) function (line 4531)
+
+        // Test data: Create a simple tree to serialize (with case insensitive flag)
```

Comment:
apart from comments, I don't see any reference to case insensitive ..
Also, that's somewhat conflating the purpose of this test.

Why not use one of the existing tree factories e.g. createCompatibilityConfigPropertyTree() for this test.

Or, it's good to keep the verbosity down... really all this new test wants to test is the new deserialize.
performRoundTripTest is already serializing (like this new test is), so you could add the new deserialize tests to it in a 2-3 lines only.